File: src/main/java/no/nav/domain/pensjon/regler/grunnlag/Sivilstand.java
package no.nav.domain.pensjon.regler.grunnlag;

import no.nav.domain.pensjon.regler.PenPerson;
import no.nav.domain.pensjon.regler.kode.GrunnlagKildeCti;
import no.nav.domain.pensjon.regler.kode.SivilstandTypeCti;

import java.io.Serializable;
import java.util.Date;

public class Sivilstand implements Serializable {



    /**
     * Kode som angir hvilken type sivilstand personen har.
     */
    private SivilstandTypeCti sivilstandType;

    /**
     * Sivilstandens gyldighet fra-og-med dato.
     */
    private Date fom;

    /**
     * Sivilstandens gyldighet til-og-med dato
     */
    private Date tom;

    /**
     * Person sivilstanden refererer seg til, for eksempel ektefellen hvis <em>sivilstandType</em>="Ektefelle"
     */
    private PenPerson relatertPerson;

    /**
     * Angir sivilstandens kilde.
     */
    private GrunnlagKildeCti kilde;

}

File: src/main/kotlin/no/nav/pensjon/regler/domain/grunnlag/Sivilstand.kt
package no.nav.pensjon.regler.domain.grunnlag

import no.nav.pensjon.regler.domain.PenPerson
import no.nav.pensjon.regler.domain.enum.GrunnlagkildeEnum
import no.nav.pensjon.regler.domain.enum.SivilstandEnum
import java.io.Serializable
import java.util.*
import java.time.LocalDate

class Sivilstand : Serializable {
    /**
     * Kode som angir hvilken type sivilstand personen har.
     */
    var sivilstandTypeEnum: SivilstandEnum? = null

    /**
     * Sivilstandens gyldighet fra-og-med dato.
     */
    @Deprecated("Use fomLd instead")
    var fom: Date? = null
    var fomLd: LocalDate? = null

    /**
     * Sivilstandens gyldighet til-og-med dato
     */
    @Deprecated("Use tomLd instead")
    var tom: Date? = null
    var tomLd: LocalDate? = null

    /**
     * Person sivilstanden refererer seg til, for eksempel ektefellen hvis *sivilstandType*="Ektefelle"
     */
    var relatertPerson: PenPerson? = null

    /**
     * Angir sivilstandens kilde.
     */
    var kildeEnum: GrunnlagkildeEnum? = null

}